eth/ethconfig/config.go

```diff
@@ -213,25 +213,23 @@ func CreateConsensusEngine(stack *node.Node, chainConfig *params.ChainConfig, co
 	switch config.PowMode {
 	case ethash.ModeFake:
 		log.Warn("Ethash used in fake mode")
-		return ethash.NewFaker()
 	case ethash.ModeTest:
 		log.Warn("Ethash used in test mode")
-		return ethash.NewTester(nil, noverify)
 	case ethash.ModeShared:
 		log.Warn("Ethash used in shared mode")
-		return ethash.NewShared()
```

Comment:
The idea here is to use a shared instance, but that's just ignored in these changes...? 
```
func NewShared() *Ethash {
	return &Ethash{shared: sharedEthash}
}
```

Reply:
This is now fixed as well. The main constructor, `New`, now sets `shared` when `ModeShared` is used.